File: problems/palindrome-number/Solution9.java
public class Solution9 {
  public static class ReverseString {
    public static boolean isPalindrome(int x) {
      final String s = String.valueOf(x);
      return s.contentEquals(new StringBuilder(s).reverse());
    }
  }

  public static class CompareHalfString {
    public static boolean isPalindrome(int x) {
      final String s = String.valueOf(x);
      int start = 0;
      int end = s.length() - 1;
      while (start < end) {
        if (s.charAt(start++) != s.charAt(end--)) {
          return false;
        }
      }
      return true;
    }
  }
}

File: problems/palindrome-number/solution9.py
class ReverseString:
    @staticmethod
    def is_palindrome(x: int) -> bool:
        return str(x) == str(x)[::-1]


class CompareHalfString:
    @staticmethod
    def is_palindrome(x: int) -> bool:
        s = str(x)
        start = 0
        end = len(s) - 1
        while start < end:
            if s[start] != s[end]:
                return False
            start += 1
            end -= 1
        return True
